src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatibilityAnalyzer.OperationVisitor.cs

```diff
@@ -43,7 +43,7 @@ private sealed class OperationVisitor : GlobalFlowStateDataFlowOperationVisitor
                         for (var i = 0; i < infosBuilder.Count; i++)
                         {
                             var newValue = new GlobalFlowStateAnalysisValueSet(infosBuilder[i]);
-                            value = i == 0 ? newValue : new GlobalFlowStateAnalysisValueSet(value, newValue);
+                            value = (i == 0 || value == newValue) ? newValue : new GlobalFlowStateAnalysisValueSet(value, newValue);
```

Comment:
❔ Why `==` instead of this?

```suggestion
                            value = (i == 0 || Equals(value, newValue)) ? newValue : new GlobalFlowStateAnalysisValueSet(value, newValue);
```